src/gameLogic/execution/gadget/Cocktail.cpp

```diff
@@ -79,5 +101,26 @@ namespace spy::gameplay {
 
         return ret;
     }
+
+    bool GadgetExecutor::executeCocktail(State &s, const GadgetAction &a, const MatchConfig &config) {
+        auto sourceChar = s.getCharacters().getByUUID(a.getCharacterId());
+        auto &targetField = s.getMap().getField(a.getTarget());
+        bool drink = (sourceChar->getCoordinates().value() == a.getTarget());
+
+        if (drink) {
+            // drink the cocktail
+            return executeDrink(s, config, sourceChar);
+        } else if (targetField.getFieldState() == scenario::FieldStateEnum::BAR_TABLE
+                   && targetField.getGadget().has_value()) {
+            // take the cocktail from the bar table
+            sourceChar->addGadget(targetField.getGadget().value());
+            targetField.removeGadget();
+
+            return true;
+        } else {
+            // pour the cocktail over another character
+            return executeCocktailAttack(s, a, config, sourceChar);
```

Comment:
hier sollte noch geprüft werden ob man einen cocktail hat, oder?

Reply:
Dieser Fall sollte [hier](https://github.com/SoPra-Team-17/LibCommon/blob/40e84be67dfe1659a6ce50905c0fbae008e13887/src/gameLogic/validation/gadget/Cocktail.cpp#L32) bereits in der validate abgefangen werden.